### ISSUE TITLE
Add a changelog

### ISSUE THREAD

**Issue by networkimprov** [blevesearch/bleve#1412]
Could the project keep a changelog in the repo, preferably in Markdown format?

There have been 11 releases, beginning with 0.8.2, without release notes. (Twelve, if you count v1.0.6, which is missing.)

I need to know what's been fixed and added in each release! Ideally with refs to the associated commits and issues for each item.

Thanks & Regards :-)

**Comment by mschoch** [comment]
I agree having changelog information would be helpful.  This project has only had actual releases for a few months, so we're still figuring things out.

My personal preference is to maintain the information through github, instead of a separate file inside the repository.  I have pulled together the information for usable `v1.0.x` releases on the project [releases](https://github.com/blevesearch/bleve/releases) page.

**Comment by networkimprov** [comment]
Thanks for adding notes to the releases!

Some reasons to keep the original notes in a changelog in-repo:
a) anyone posting a PR can update the changelog within a fix/enhancement patch,
b) folks with a git clone of the repo have the changelog locally
c) the event that you move out of Github one day.
